File: _includes/sort-benchmark/algorithms/brick.rs
fn brick_sort(a: &mut [usize]) {
    let mut sorted = false;
    while !sorted {
        sorted = true;
        for i in (1..a.len()).step_by(2) {
            if a[i - 1] > a[i] {
                a.swap(i - 1, i);
                sorted = false;
            }
        }
        for i in (2..a.len()).step_by(2) {
            if a[i - 1] > a[i] {
                a.swap(i - 1, i);
                sorted = false;
            }
        }
    }
}

File: _includes/sort-benchmark/algorithms/brick.swift
func brick_sort(_ a: inout [Int]) {
    a.withUnsafeMutableBufferPointer { brick_sort($0) }
}

func brick_sort(_ a: UnsafeMutableBufferPointer<Int>) {
    var sorted = false
    while !sorted {
        sorted = true
        var i = 1
        while i < a.count {
            if a[i - 1] > a[i] {
                a.swapAt(i - 1, i)
                sorted = false
            }
            i += 2
        }
        i = 2
        while i < a.count {
            if a[i - 1] > a[i] {
                a.swapAt(i - 1, i)
                sorted = false
            }
            i += 2
        }
    }
}
